crates/red_knot_python_semantic/src/types/type_ordering.rs

```diff
@@ -11,16 +11,21 @@ use super::{
 /// in an [`crate::types::IntersectionType`] or a [`crate::types::UnionType`] in order for them
 /// to be compared for equivalence.
 ///
-/// Two unions with equal sets of elements will only compare equal if they have their element sets
-/// ordered the same way.
+/// Two unions / intersections are compared lexicographically.
+///
+/// Element types must already be sorted.
 ///
 /// ## Why not just implement [`Ord`] on [`Type`]?
 ///
 /// It would be fairly easy to slap `#[derive(PartialOrd, Ord)]` on [`Type`], and the ordering we
 /// create here is not user-facing. However, it doesn't really "make sense" for `Type` to implement
 /// [`Ord`] in terms of the semantics. There are many different ways in which you could plausibly
 /// sort a list of types; this is only one (somewhat arbitrary, at times) possible ordering.
-pub(super) fn union_elements_ordering<'db>(left: &Type<'db>, right: &Type<'db>) -> Ordering {
+pub(super) fn union_elements_ordering<'db>(
```

Comment:
Orthogonal to this PR, but this method is used to sort elements of both unions and intersections. The doc comment above describes this correctly, but this name is misleading. I would suggest we rename to `union_or_intersection_elements_ordering`. (It only has two call sites, so verbosity is not really an issue.)